### DISCUSSION TITLE
Sending Emails Using an Email Template

### DISCUSSION THREAD

**Discussion by IRodriguezSC** [sadieblue/agilityblue.workspace-scripts#23] [Q&A] [answered]
Hi, I'm trying to kick-off an email from an email template but not sure how to get this done.  I'm able to get the email template but how do I sending it.

**Comment by bmward80** [comment, marked as answer]
Hi @IRodriguezSC,

Are you trying to kick off an email when a specific event happens, or are you trying to execute a script manually? Generally speaking, if you are using an email template, and you have tokens defined in it, the email should only be kicked off from some kind of task-related event, otherwise your tokens will contain no value.

To send an email, you will first need to create one using the [Add-EmailMessage](https://help.agilityblue.com/docs/scripting-add-emailmessage) cmdlet. Then you will need to add email message contacts using the [Add-EmailMessageContacts](https://help.agilityblue.com/docs/scripting-add-emailmessagecontacts) cmdlet. If you have any attachments, you can use the [Add-EmailMessageAttachments](https://help.agilityblue.com/docs/scripting-add-emailmessageattachments) cmdlet. Then, you will link the email message to a task using the [Add-TaskEmailMessage](https://help.agilityblue.com/docs/scripting-add-taskemailmessage) cmdlet, and finally, you can send the email using the [Send-EmailMessage](https://help.agilityblue.com/docs/scripting-send-emailmessage) cmdlet.

For some examples:

Creating an email messge:

```powershell
$emailMessage = @{
    To          = "someone@somewhere.com"
    FromAddress = "you@atyourcompany.com"
    FromName    = "Your Name"
    Subject     = "Email created from a a script"
    Body        = "<p>This is an email created from a script.</p><p>I hope you have a great day!</p>"
}

$createdEmailMessage = Add-EmailMessage $emailMessage
```

Adding contacts to an email:

```powershell
# Relationship Types:
# ----------------------------------------------
# 1 = To
# 2 = From
# 3 = CC
# 4 = BCC
# 5 = Linked
# ----------------------------------------------

$emailMessageContacts = @(
  @{
    EmailMessageId     = 12805 # an email that was previously created
    ContactId          = 2 # a contact that already exists
    RelationshipTypeId = 1
  }
)

Add-EmailMessageContacts -Entries $emailMessageContacts | Out-Null
```

Creating a task email and sending it:

```powershell
# In order to send an email, the email and task must first be created, and then a link between them must be created.

$taskEmailMessage = @{
  TaskId         = 29877 # a task that was previously created
  EmailMessageId = 12669 # an email that was previously created
}

Add-TaskEmailMessage $taskEmailMessage | Out-Null
Send-EmailMessage 12669 | Out-Null
```

I hope this helps,
Brandon

**Comment by IRodriguezSC** [comment]
Thank you Brandon!  The email will be kicked-off as a result of a task-related event.   The example you provided are clear and I will give it a try today.

Thanks,
-IR

From: Brandon Ward ***@***.***>
Sent: Friday, August 30, 2024 8:21 AM
To: sadieblue/agilityblue.workspace-scripts ***@***.***>
Cc: Rodriguez, Ismael ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [sadieblue/agilityblue.workspace-scripts] Sending Emails Using an Email Template (Discussion #23)


Hi @IRodriguezSC [github.com]<https://urldefense.com/v3/__https:/github.com/IRodriguezSC__;!!KhposeJF9Q!SYOQ0smcezB9mfG9yA1-eT2Jt3A80BvHnMQWzVZeh79DftCa_4u65W_A2n44NJOd0jK6oyrU0bEUfYtOThae_2RiZ_8$>,

Are you trying to kick off an email when a specific event happens, or are you trying to execute a script manually? Generally speaking, if you are using an email template, and you have tokens defined in it, the email should only be kicked off from some kind of task-related event, otherwise your tokens will contain no value.

To send an email, you will first need to create one using the Add-EmailMessage [help.agilityblue.com]<https://urldefense.com/v3/__https:/help.agilityblue.com/docs/scripting-add-emailmessage__;!!KhposeJF9Q!SYOQ0smcezB9mfG9yA1-eT2Jt3A80BvHnMQWzVZeh79DftCa_4u65W_A2n44NJOd0jK6oyrU0bEUfYtOThae_Fu-_pM$> cmdlet. Then you will need to add email message contacts using the Add-EmailMessageContacts [help.agilityblue.com]<https://urldefense.com/v3/__https:/help.agilityblue.com/docs/scripting-add-emailmessagecontacts__;!!KhposeJF9Q!SYOQ0smcezB9mfG9yA1-eT2Jt3A80BvHnMQWzVZeh79DftCa_4u65W_A2n44NJOd0jK6oyrU0bEUfYtOThaeoV_OJjk$> cmdlet. If you have any attachments, you can use the Add-EmailMessageAttachments [help.agilityblue.com]<https://urldefense.com/v3/__https:/help.agilityblue.com/docs/scripting-add-emailmessageattachments__;!!KhposeJF9Q!SYOQ0smcezB9mfG9yA1-eT2Jt3A80BvHnMQWzVZeh79DftCa_4u65W_A2n44NJOd0jK6oyrU0bEUfYtOThaeMT0Rdds$> cmdlet. Then, you will link the email message to a task using the Add-TaskEmailMessage [help.agilityblue.com]<https://urldefense.com/v3/__https:/help.agilityblue.com/docs/scripting-add-taskemailmessage__;!!KhposeJF9Q!SYOQ0smcezB9mfG9yA1-eT2Jt3A80BvHnMQWzVZeh79DftCa_4u65W_A2n44NJOd0jK6oyrU0bEUfYtOThaeJVrBXu4$> cmdlet, and finally, you can send the email using the Send-EmailMessage [help.agilityblue.com]<https://urldefense.com/v3/__https:/help.agilityblue.com/docs/scripting-send-emailmessage__;!!KhposeJF9Q!SYOQ0smcezB9mfG9yA1-eT2Jt3A80BvHnMQWzVZeh79DftCa_4u65W_A2n44NJOd0jK6oyrU0bEUfYtOThae-AyBJto$> cmdlet.

For some examples:

Creating an email messge:

$emailMessage = @{

    To          = ***@***.******@***.***>"

    FromAddress = ***@***.******@***.***>"

    FromName    = "Your Name"

    Subject     = "Email created from a a script"

    Body        = "<p>This is an email created from a script.</p><p>I hope you have a great day!</p>"

}



$createdEmailMessage = Add-EmailMessage $emailMessage

Adding contacts to an email:

# Relationship Types:

# ----------------------------------------------

# 1 = To

# 2 = From

# 3 = CC

# 4 = BCC

# 5 = Linked

# ----------------------------------------------



$emailMessageContacts = @(

  @{

    EmailMessageId     = 12805 # an email that was previously created

    ContactId          = 2 # a contact that already exists

    RelationshipTypeId = 1

  }

)



Add-EmailMessageContacts -Entries $emailMessageContacts | Out-Null

Creating a task email and sending it:

# In order to send an email, the email and task must first be created, and then a link between them must be created.



$taskEmailMessage = @{

  TaskId         = 29877 # a task that was previously created

  EmailMessageId = 12669 # an email that was previously created

}



Add-TaskEmailMessage $taskEmailMessage | Out-Null

Send-EmailMessage 12669 | Out-Null

I hope this helps,
Brandon

—
Reply to this email directly, view it on GitHub [github.com]<https://urldefense.com/v3/__https:/github.com/sadieblue/agilityblue.workspace-scripts/discussions/23*discussioncomment-10498450__;Iw!!KhposeJF9Q!SYOQ0smcezB9mfG9yA1-eT2Jt3A80BvHnMQWzVZeh79DftCa_4u65W_A2n44NJOd0jK6oyrU0bEUfYtOThaeWNty2-A$>, or unsubscribe [github.com]<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/BK4IHLBFAUAAIYRCIHBYOMDZUBPUDAVCNFSM6AAAAABNK5PKB6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBZHA2DKMA__;!!KhposeJF9Q!SYOQ0smcezB9mfG9yA1-eT2Jt3A80BvHnMQWzVZeh79DftCa_4u65W_A2n44NJOd0jK6oyrU0bEUfYtOThaePe-dKIM$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

________________________________
**This is an external message from: ***@***.******@***.***> **

________________________________
This e-mail is sent by a law firm and contains information that may be privileged and confidential. If you are not the intended recipient, please delete the e-mail and notify us immediately.
